src/api/users.ts

```diff
@@ -32,3 +32,47 @@ router.delete("/users/history/:courseId", async (req, res) => {
   await sql`DELETE FROM courses_completed WHERE user_id = ${user.id} AND course_id = ${courseId}`;
   res.send();
 });
+
+router.get("/users/progress/:planId", async (req, res) => {
+  const user = await authUser(req);
+  const planId = req.params.planId;
+
+  const result = await sql.begin(async (sql) => {
+    const planDetails = await sql`
+      SELECT pr.total_units, pr.required_courses
+      FROM plans p
+      JOIN programs pr ON pr.id = p.program_id
+      WHERE p.id = ${planId} AND p.user_id = ${user.id}`;
+
+    const { totalUnits, requiredCourses } = planDetails[0];
+
+    const completedCourses = await sql`
+      SELECT COUNT(*) AS completed_count
+      FROM courses_completed cc
+      WHERE cc.user_id = ${user.id} AND cc.course_id = ${requiredCourses}`;
+
+    const completedCount = completedCourses[0].completedCount;
```

Comment:
Rather than doing `[0]` on here, I'd prefer if we use the `one()` utility method

See courses:
https://github.com/DegreePrerequisiteTracker/Degree-Prerequisite-Tracker/blob/main/src/api/courses.ts#L22